internal/cmd/egctl/envoy_stats.go

```diff
@@ -0,0 +1,200 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/spf13/cobra"
+	"github.com/tetratelabs/multierror"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+const (
+	summaryOutput    = "statsd"
+	prometheusOutput = "prom"
+)
+
+var (
+	statsType, outputFormat string
+)
+
+func newEnvoyStatsConfigCmd() *cobra.Command {
+	var podName, podNamespace string
+
+	statsConfigCmd := &cobra.Command{
+		Use:   "envoy-stats [<type>/]<name>[.<namespace>]",
```

Comment:
can we follow the same UX as `egctl config` ?
https://github.com/envoyproxy/gateway/blob/b98c1d1cc88686765128a0228f9acbed14e7c26f/internal/cmd/egctl/config_cmd.go#L61
so this would look like
```
egctl stats envoy-proxy all <po> -n <ns>
```

Reply:
its a little more wordy, but it allows us to easier add stats for `envoy-ratelimit` and `envoy-gateway` ?

Reply:
stats is not a kind of config, prefer to keep it separated.

Reply:
@zirain can you please rephrase ?
are you saying that
```egctl stats envoy-proxy``` | ```egctl stats envoy-ratelimit``` | ```egctl stats envoy-gateway```
dont make much sense ?

Reply:
sorry, I miss understand what you said.

yes, `egctl stats` is better.

Reply:
@arkodg @zirain the command now looks like `egctl stats envoy-proxy  <po> -n <ns>`. I don't think there is any need for all in this command. wdyt?

Reply:
yes, it's OK to me.

Reply:
It is exposed through `--`. It is same for istio. Not saying it should be the same way but the command is pretty much the same, so from a user perspective it's better in my opinion.